haskeline.cabal

```diff
@@ -20,7 +20,7 @@ Homepage:       http://trac.haskell.org/haskeline
 Bug-Reports:    https://github.com/judah/haskeline/issues
 Stability:      Stable
 Build-Type:     Custom
-extra-source-files: examples/Test.hs Changelog
+extra-source-files: examples/Test.hs Changelog includes/*.h
```

Comment:
Sure. So when I originally started working on this PR, I noticed that there were two files with this exact boilerplate at the top:

```haskell
##if defined(i386_HOST_ARCH)
## define WINDOWS_CCONV stdcall
##elif defined(x86_64_HOST_ARCH)
## define WINDOWS_CCONV ccall
##else
## error Unknown mingw32 arch
##endif
```

And I was going to be adding a third file with the same boilerplate! So I moved this logic into `includes/windows_cconv.h`.

Note that `win_console.h` is put in an `install-includes` stanza of `haskeline.cabal`, which means that any library which transitively depends on `haskeline` can use `#include "win_console.h"` in their code. But `windows_cconv.h` is purely internal to `haskeline`, so I didn't put it in `install-includes`.

Sadly, for whatever reason `Cabal` doesn't put any files under an `includes` stanza into the tarball when you run `cabal sdist` (only the ones under `install-includes`). So to work around this, you have to list these files yourself in `extra-source-files`, which _are_ put into the tarball on a `cabal sdist`.

Alternatively, I could just remove `windows_cconv.h` entirely and go back to the way it was before, it you'd prefer that.